core/capabilities/remote/target/caller.go

```diff
@@ -0,0 +1,142 @@
+package target
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	commoncap "github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/target/request"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// caller is a shim for remote target capabilities.
+// It translates between capability API calls and network messages.
+// Its responsibilities are:
+//  1. Transmit capability requests to remote nodes according to a transmission schedule
+//  2. Aggregate responses from remote nodes and return the aggregated response
+//
+// caller communicates with corresponding receiver on remote nodes.
+type caller struct {
+	lggr                 logger.Logger
+	remoteCapabilityInfo commoncap.CapabilityInfo
+	localDONInfo         capabilities.DON
+	dispatcher           types.Dispatcher
+	requestTimeout       time.Duration
+
+	messageIDToCallerRequest map[string]*request.CallerRequest
+	mutex                    sync.Mutex
+}
+
+var _ commoncap.TargetCapability = &caller{}
+var _ types.Receiver = &caller{}
+
+func NewCaller(ctx context.Context, lggr logger.Logger, remoteCapabilityInfo commoncap.CapabilityInfo, localDonInfo capabilities.DON, dispatcher types.Dispatcher,
+	requestTimeout time.Duration) *caller {
+	c := &caller{
+		lggr:                     lggr,
+		remoteCapabilityInfo:     remoteCapabilityInfo,
+		localDONInfo:             localDonInfo,
+		dispatcher:               dispatcher,
+		requestTimeout:           requestTimeout,
+		messageIDToCallerRequest: make(map[string]*request.CallerRequest),
+	}
+
+	go func() {
+		ticker := time.NewTicker(requestTimeout)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				c.ExpireRequests()
+			}
+		}
+	}()
+
+	return c
+}
+
+func (c *caller) ExpireRequests() {
```

Comment:
Does this need to be public? AFAICS it's only called by the worker goroutine

Reply:
(Same comment for the Receiver)

Reply:
ta, sorted that